modules/renter/download.go

```diff
@@ -183,11 +160,40 @@ func (r *Renter) Download(nickname, destination string) error {
 	if err != nil {
 		return err
 	}
-	go d.start()
 
 	// Add the download to the download queue.
 	r.downloadQueue = append(r.downloadQueue, d)
-	return nil
+	r.mu.Unlock(lockID)
+
+	// Download the file. We only need one piece, so iterate through the hosts
+	// until a download succeeds.
+	for i := 0; i < downloadAttempts; i++ {
+		for _, piece := range d.pieces {
+			downloadErr := d.downloadPiece(piece)
+			if downloadErr == nil {
+				// done
+				d.complete = true
+				d.file.Close()
+				return nil
+			}
+			// Reset seek, since the file may have been partially written. The
+			// next attempt will overwrite these bytes.
+			d.file.Seek(0, 0)
+			atomic.SwapUint64(&d.received, 0)
```

Comment:
what we would really want is to start the next download from the same place the failure left off. I think we will get that after the revamp